README.md

```diff
@@ -1,4 +1,49 @@
-# sha256-as
-Assembly Script implementation of SHA256
-Built using the AssemblyScript Dev Branch
-To run, `node run.js`.
\ No newline at end of file
+# as-sha256
+Assembly Script implementation of SHA256.
+
+This repository contains two npm distributions; a pure AssemblyScript version, and a Typescript wrapper for the WASM build.
+
+## Benchmarks
+
+```
+Sha256-Rust#hash x 328,611 ops/sec ±5.99% (75 runs sampled)
+Sha256-Js#hash x 340,641 ops/sec ±4.64% (77 runs sampled)
+Sha256-Asm#hash x 5,217 ops/sec ±8.83% (74 runs sampled)
+Sha256-BCrypto#hash x 411,191 ops/sec ±9.87% (59 runs sampled)
+AS-Sha256#hash x 537,124 ops/sec ±8.91% (76 runs sampled)
+Fastest is AS-Sha256#hash
+```
+
+
+## AssemblyScript
+
+`yarn add @chainsafe/as-sha256`
+
+## Typescript Warpper
+
+`yarn add @chainsafe/sha256`
+
+## Usage
+
+```JS
+import sha256 from "@chainsafe/sha256";
+
+const message = Buffer.from("Hello world");
+
+const hash = sha256(message);
```

Comment:
Our most common use case is init once, doing updates in loops and finish. So I would suggest 
in typescript wrapper create class called Sha256 where in constructor you would call wasm init method and expose update and finish as methods. Class could also have static method hash.

Reply:
I can't seem to find typescript wrapper code

Reply:
I'm making a declaration file tomorrow !

Pr is a wip